os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -77,6 +77,51 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+static void allocateregions(FAR struct elf_loadinfo_s *loadinfo)
+{
+	size_t sizes[3] = {loadinfo->textsize, loadinfo->rosize, loadinfo->binp->ramsize};
+	uintptr_t *allocs[3] = {&loadinfo->textalloc, &loadinfo->roalloc, &loadinfo->dataalloc};
+	uint32_t totalsize = sizes[0] + sizes[1] + sizes[2];
+	size_t tmpsz;
+	uintptr_t *tmpalloc;
+
+
+	if (sizes[0] < sizes[1]) {
+		tmpsz = sizes[0];
+		sizes[0] = sizes[1];
+		sizes[1] = tmpsz;
+
+		tmpalloc = allocs[0];
+		allocs[0] = allocs[1];
+		allocs[1] = tmpalloc;
+	}
+
+	if (sizes[1] < sizes[2]) {
+		tmpsz = sizes[1];
+		sizes[1] = sizes[2];
+		sizes[2] = tmpsz;
+
+		tmpalloc = allocs[1];
+		allocs[1] = allocs[2];
+		allocs[2] = tmpalloc;
+
+		if (sizes[0] < sizes[1]) {
+			tmpsz = sizes[0];
+			sizes[0] = sizes[1];
+			sizes[1] = tmpsz;
+
+			tmpalloc = allocs[0];
+			allocs[0] = allocs[1];
+			allocs[1] = tmpalloc;
+		}
+	}
+
+	loadinfo->binp->ramstart = *allocs[0] = (uintptr_t)kumm_memalign(sizes[0], totalsize);
+	*allocs[1] = *allocs[0] + sizes[0];
+	*allocs[2] = *allocs[1] + sizes[1];
+}
```

Comment:
Could we free the pointer array `allocs` and `tmpalloc`

Reply:
We dont have to free them since they are static arrays and not allocated. And the values stored in alloc array are used by the loader in future.

Reply:
`allocs` is local array. How can loader use this...?

Reply:
allocs is a pointer to pointer.